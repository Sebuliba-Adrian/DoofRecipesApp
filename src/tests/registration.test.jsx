import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import axios from 'axios';
import { mount, configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Registration from '../components/authentication/registration';
configure({ adapter: new Adapter() });

const props = {
  location: {
    data: {
      message: 'Test message',
    }
  },
  history: {
    replace: jest.fn()
  }
};

describe('Test registration', () => {
  const wrapper = mount(<Registration {...props} />);
  const instance = wrapper.instance();

  global.fetch = jest.fn(() => Promise.rejects({}));
  it('renders correctly', () => {
    const registration = renderer.create(<Registration {...props} />).toJSON();
    expect(registration).toMatchSnapshot();
  });

  it('submits the form', () => {
    
    const userName = wrapper.find('#username');
    userName.simulate('change', {
      target: { name: 'username', value: 'testusername' }
    });
    userName.value = 'testusername';
    expect(userName.value).toEqual('testusername');

    const userEmail = wrapper.find('#email');
    userEmail.simulate('change', {
      target: { name: 'email', value: 'test@test.com' }
    });
    userEmail.value = 'test@test.com';
    expect(userEmail.value).toEqual('test@test.com');

    const userPassword = wrapper.find('#password');
    const confUserPassword = wrapper.find('#confirm_password');
    userPassword.simulate('change', {
      target: { name: 'password', value: 'testpassword"'}
    });
    userPassword.simulate('change', {
      target: { name: 'confm_password', value: 'testpassword' }
    });
    userPassword.value = 'testpassword';
    expect(userPassword.value).toEqual('testpassword');

    
    
    
    const form = wrapper.find('form');
    form.find('button').simulate('submit', { preventDefault: () => {} });
  });

  
});
