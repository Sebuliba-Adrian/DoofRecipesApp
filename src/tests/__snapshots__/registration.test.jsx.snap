// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test registration renders correctly 1`] = `
<div
  className="App "
>
  <div
    className="row"
  >
    <div
      className="col-md-4 offset-md-4 col-xs-10 offset-xs-2 "
    >
      <div
        className="card mt-5 p-5"
      >
        <div
          className="card-block"
        >
          <div>
            <h3
              className="logo text-center mb-3"
            >
              Doof Recipes App
            </h3>
            <p
              className="grey-text text-center mb-5"
            >
              Record your favourite recipes 
            </p>
          </div>
          <div
            className="line-with-text font-weight-bold mb-4"
          >
            <div
              className="line"
            />
            <div
              className="line-text"
            >
              REGISTER
            </div>
            <div
              className="line"
            />
          </div>
          <form
            onSubmit={[Function]}
          >
            <input
              className="form-control mb-1"
              id="username"
              name="username"
              onChange={[Function]}
              placeholder="Username"
              required={true}
              type="text"
              value=""
            />
            <input
              className="form-control mb-1"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="email"
              value=""
            />
            <input
              className="form-control mb-1"
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
              value=""
            />
            <input
              className="form-control mb-1"
              id="confirm_password"
              name="confm_password"
              onChange={[Function]}
              placeholder="Confirm password"
              required={true}
              type="text"
              value=""
            />
            <button
              className="btn btn-primary btn-sm col-md-12"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
      <div
        className="card mt-2"
      >
        <div
          className="card-block"
        >
          <p
            className="card-text text-center message"
          >
            Have an account? 
            <a
              className="card-link"
              href="/login"
            >
              Log in
            </a>
            <a
              className="card-link"
              href="/instructions"
            >
              How To
            </a>
          </p>
        </div>
      </div>
      <div
        id="snackbar"
      />
    </div>
  </div>
</div>
`;
